### ISSUE TITLE
Queue Trigger > Allow 'none' encoding on a single function Queue Trigger

### ISSUE THREAD

**Issue by ComtelJeremy** [Azure/azure-functions-dotnet-worker#2627]
### Description

We have an outside service that is able to push directly to our Azure Queue, which eliminates the need for us to run a seperate HTTP function to receive messages. However, they do not base64 encode their message. It would be helpful to allow a single queue trigger to receive a non-encoded message (rather than changing the host.json and changing it for the entire function app). Is this possible, or am I perhaps missing some feature that already exists?

**Comment by sravanreddy6745** [comment]
Do you want the queue trigger function to receive the encoded message?

**Comment by ComtelJeremy** [comment]
No, we would like the queue trigger to receive the un-encoded message. We want the rest of the functions in the app to use base64 encoding, as is standard.

**Comment by liliankasem** [comment]
Thanks for opening this issue.

This is a valid feature request. Currently, messageEncoding can only be configured at the app level via host.json, which applies to all queue triggers in your function app. There's no mechanism today to override this on a per-function basis.

Current workarounds:

- Use a separate function app for the queue receiving non-encoded messages
- Have your external service base64-encode the messages before pushing to the queue

Path forward:

This feature would require changes in the underlying WebJobs Storage Queues extension to support per-trigger encoding configuration before we can expose it in the isolated worker extension. The binding infrastructure currently applies `QueuesOptions.MessageEncoding` globally to all queue bindings.

I'll add this to our backlog to track. In the meantime, if this is a high-priority scenario for you, I'd encourage you to also open an issue in the [azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net) repo to request this capability in the underlying extension, as that's where the core change would need to happen first.
